Solver_interface/include/CGAL/Diagonalize_traits.h

```diff
@@ -26,9 +26,11 @@
 #include <CGAL/number_type_config.h>
 #include <CGAL/double.h>
 
+#ifndef CGAL_I_WANT_TO_USE_DIAGONALIZE_TRAITS
 #define CGAL_WARNING_DIAGONALIZE_TRAITS \
   CGAL_DEPRECATED_MSG("CGAL::Diagonalize_traits is a deprecated class that can \
 lead to precision issues, please use CGAL::Eigen_diagonalize_traits")
+#endif
```

Comment:
You have not tested your patch. There must be an `#else` that defines the macro to an empty value.